device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -151,7 +151,18 @@ else if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.X509
                     clientId += "/" + moduleId;
                 }
 
-                this.iotHubUserName = this.config.getIotHubHostname() + "/" + clientId + "/" + API_VERSION + "&" + clientUserAgentIdentifier;
+                String apiVersion;
```

Comment:
rename this to serverParam or something similar, this has both api version and model Id (for preview APIs)

Reply:
I know. I though about a better naming! I would be looking any other suggestions too other than serverParam. :)

Reply:
An option to consider might be - Define two variabes - apiVersion, modelIdParam and initialize both inside the if-else. When you do str concatenation later, it should just work (if it works like c#)
this.iotHubUserName = this.config.getIotHubHostname() + "/" + clientId + "/?api-version=" + apiVersion + modelIdParam

Reply:
I like your suggestion. I also need to check how it works in case where modelId is not part of username. Currently I have it renamed as per Abhipsa's comment